models/account/account.go

```diff
@@ -0,0 +1,106 @@
+package account
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/b0d0nne11/scroll-test-project/db"
+	"github.com/mailgun/scroll"
+)
+
+type Account struct {
+	ID   int64
+	Name string
+}
+
+func New(name string) *Account {
+	return &Account{
+		Name: name,
+	}
+}
+
+func (a *Account) Save() error {
```

Comment:
I understand how there could be some threading issues here, esp with regard to the ID value but I thought that the atomic nature of the SQL transactions would make this mostly safe. That probably wont help as soon as I move it over to MongoDB so I'd need to add some thread safety anyway. What would you suggest? Add a lock around the save portion with a mutex?